QUIZ APP/quiz.js

```diff
@@ -0,0 +1,228 @@
+    //             quizSection.style.display = "block";
+    //             startQuiz();
+    //         } else {
+    //             alert("Please enter a name for the player.");
+    //         }
+    //     }
+    
+    
+    function startQuiz(){
+        currentQuestionIndex = 0;
+        currentPlayerIndex = 0;
+        score = [0];
+        nextButton.innerHTML = "Next";
+        shuffleButton.innerHTML = "Shuffle Answers";
+        shuffleButton.style.display = "block";
+        showQuestion();
+    }
+    
+    function showQuestion(){
+        resetState();
+        let currentQuestion = questions[currentQuestionIndex];
+        let questionNumber = currentQuestionIndex + 1;
+        questionElement.innerHTML = questionNumber + ". " + currentQuestion.question;
+       
+        
+        currentQuestion.answers.forEach(answer => {
+            const button = document.createElement("button");
+            button.innerHTML =  answer.text;
+            button.classList.add("btn");
+            answerButtons.appendChild(button);
+    
+            if(answer.correct){
+                button.dataset.correct = answer.correct;
+           }
+            button.addEventListener("click", selectAnswer);
+        });
+    };
+    
+    function resetState(){
+        nextButton.style.display = "none";
+        explanationElement.innerHTML = "";
+        while(answerButtons.firstChild){
+            answerButtons.removeChild(answerButtons.firstChild);
+        }
+    }
+    
+    function selectAnswer(e){
+        const selectButton = e.target;
+        const isCorrect = selectButton.dataset.correct === "true";
+    
+        if(isCorrect){
+            selectButton.classList.add("correct");
+            score[currentPlayerIndex]++;
+        } else {
+            selectButton.classList.add("incorrect");
+        }
+    
+        Array.from(answerButtons.children).forEach(button => {
+            if(button.dataset.correct === "true"){
+                button.classList.add("correct");
+            }
+            button.disabled = true;
+        });
+    
+        explanationElement.innerHTML = questions[currentQuestionIndex].explanation;
+        nextButton.style.display = "block";
+    
+    }
+    
+    function showScore(){
+        resetState();
+        // questionElement.innerHTML = "The score for " + players[0] +" is:" + score[0]  + "and for player " + players[1] + "is: " + score[1] + " of " + questions.length;
+        questionElement.innerHTML = "The score for: " + playerName + " is :" + score + " of " + questions.length;
+        nextButton.innerHTML = "Start the quiz again";
+        nextButton.style.display = "block";
+        shuffleButton.style.display = "none";
+    
+    }
+    
+    function goToNextButton(){
+        currentPlayerIndex = (currentPlayerIndex + 1) % 2;
+    
+        if (currentPlayerIndex === 0){
+            currentQuestionIndex++;
+        }
+    
+            if(currentQuestionIndex < questions.length){
```

Comment:
This is a good example of how formatting could help make the code easier to follow. Here, the two if's are on the same level, but visually, they are not. Just to give you an example why I insist on formatting 😊